### PR TITLE
Fetch and refresh live memory metrics for a process

### PR DESCRIPTION
**Pull request by vudayani-vmw**

### CODE REVIEW

File: headless-services/spring-boot-language-server/src/main/java/org/springframework/ide/vscode/boot/java/livehover/v2/ActuatorConnection.java [hunk 1]
```diff
@@ -31,4 +31,8 @@ public interface ActuatorConnection {
 	String getMetrics(String metric, Map<String, String> tags) throws IOException;
 
 	Map<?, ?> getStartup() throws IOException;
+
```
[the remark this answers is not in the document]
Comment (vudayani-vmw) [reply headless-services/spring-boot-language-server/src/main/java/org/springframework/ide/vscode/boot/java/livehover/v2/ActuatorConnection.java]: Removed the commented code

File: headless-services/spring-boot-language-server/src/main/java/org/springframework/ide/vscode/boot/java/livehover/v2/SpringProcessConnectorService.java [hunk 2]
```diff
@@ -99,7 +99,7 @@ public void connectProcess(String processKey, SpringProcessConnector connector)
 		}
 	}
 	
-	public void refreshProcess(String processKey) {
+	public void refreshProcess(String processKey, String endpoint, String metricName) {
```
[the remark this answers is not in the document]
Comment (vudayani-vmw) [reply headless-services/spring-boot-language-server/src/main/java/org/springframework/ide/vscode/boot/java/livehover/v2/SpringProcessConnectorService.java]: I have made the below changes to fetch process summary data for gc, memory metrics. Do these changes fit in with your suggestion? 

```
getClient().liveProcessMemoryMetricsDataUpdated(createMemoryMetricsSummary(processKey, liveData));

getClient().liveProcessGcPausesMetricsDataUpdated(createGcPausesMetricsSummary(processKey, liveData));
```

I have tried to refactor the `refreshProcess(...)` and `scheduleRefresh(...)` methods as below:
```
private CompletableFuture<Object> refresh(ExecuteCommandParams params) {
	SpringProcessParams springProcessParams = new SpringProcessParams();
        springProcessParams.setProcessKey(getProcessKey(params));    
        springProcessParams.setEndpoint(getArgumentByKey(params, "endpoint"));
		if (springProcessParams.getProcessKey() != null) {
			connectorService.refreshProcess(springProcessParams);
		}

		return CompletableFuture.completedFuture(null);
	}
	
public void refreshProcess(SpringProcessParams springProcessParams) {
....
}
``` 
Do these changes look good? should we try to refactor this more?

File: headless-services/spring-boot-language-server/src/main/java/org/springframework/ide/vscode/boot/java/livehover/v2/SpringProcessConnectorOverHttp.java [hunk 3]
```diff
@@ -102,4 +102,18 @@ public String getProcessName() {
 	public String toString() {
 		return "SpringProcessConnectorOverHttp [actuatorURL=" + actuatorUrl + "]";
 	}
+
```
[the remark this answers is not in the document]
Comment (vudayani-vmw) [reply headless-services/spring-boot-language-server/src/main/java/org/springframework/ide/vscode/boot/java/livehover/v2/SpringProcessConnectorOverHttp.java:105]: Added implementation for `refreshGcPausesMetrics` and `refreshMemoryMetrics` for connector over Http

**Comment by BoykoAlex** [comment]
I have merged this PR

**Pull request closed without merging**